tenant/service_urm_test.go

```diff
@@ -0,0 +1,50 @@
+package tenant_test
+
+import (
+	"context"
+	"testing"
+
+	"github.com/influxdata/influxdb"
+	"github.com/influxdata/influxdb/kv"
+	"github.com/influxdata/influxdb/tenant"
+	influxdbtesting "github.com/influxdata/influxdb/testing"
+)
+
+func TestBoltUserResourceMappingService(t *testing.T) {
+	influxdbtesting.UserResourceMappingService(initBoltUserResourceMappingService, t)
+}
+
+func initBoltUserResourceMappingService(f influxdbtesting.UserResourceFields, t *testing.T) (influxdb.UserResourceMappingService, func()) {
+	s, closeBolt, err := NewTestBoltStore(t)
+	if err != nil {
+		t.Fatalf("failed to create new kv store: %v", err)
+	}
+
+	svc, closeSvc := initUserResourceMappingService(s, f, t)
+	return svc, func() {
+		closeSvc()
+		closeBolt()
+	}
+}
+
+func initUserResourceMappingService(s kv.Store, f influxdbtesting.UserResourceFields, t *testing.T) (influxdb.UserResourceMappingService, func()) {
+	storage, err := tenant.NewStore(s)
+	if err != nil {
+		t.Fatal(err)
+	}
+	svc := tenant.NewService(storage)
+
+	for _, m := range f.UserResourceMappings {
+		if err := svc.CreateUserResourceMapping(context.Background(), m); err != nil {
+			t.Fatalf("failed to populate mappings")
+		}
+	}
+
+	return svc, func() {
+		for _, m := range f.UserResourceMappings {
+			if err := svc.DeleteUserResourceMapping(context.Background(), m.ResourceID, m.UserID); err != nil {
```

Comment:
I notice you mostly have tests just for create and delete, is that because find/update is covered by other tests?

Reply:
I actually didn't write any tests for these systems. Instead I am relying on the `/testing` packages test suites to make sure that these new service definitions act like the kv system (which is being tested by the same testing suite). I had to make a few updates to the testing framework because I wanted to remove the direct access some of the test required on the service implementer. But in most cases the tests are all in the testing package.